src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -427,32 +427,49 @@ private void handleLoggingConfig() {
     /**
      * Handle logging configuration changes.
      *
-     * @param what         What changed.
-     * @param loggingParam which logging param changed topic.
+     * @param what What changed.
+     * @param node which logging topic changed.
      */
     @SuppressWarnings("PMD.UselessParentheses")
-    public synchronized void handleLoggingConfigurationChanges(WhatHappened what, Node loggingParam) {
-        LoggerConfiguration configuration;
-        try {
-            configuration = fromPojo(loggingTopics.toPOJO());
-            LogManager.setEffectiveConfig(configuration);
-        } catch (IllegalArgumentException e) {
-            logger.atError().kv("logging-config", loggingTopics).cause(e).log("Unable to parse logging config.");
-            return;
-        }
-        if (currentConfiguration == null || !currentConfiguration.equals(configuration)) {
-            if (configuration.getOutputDirectory() != null
-                    && (currentConfiguration == null || !Objects.equals(currentConfiguration.getOutputDirectory(),
-                    configuration.getOutputDirectory()))) {
+    public synchronized void handleLoggingConfigurationChanges(WhatHappened what, Node node) {
+        logger.atDebug().kv("logging-change-what", what).kv("logging-change-node", node).log();
+        switch (what) {
```

Comment:
do these branches have 100% coverage in your test?

Reply:
Modified one of the unit test, now covering all three events.